subprojects/core/src/test/groovy/org/gradle/internal/fingerprint/classpath/impl/ClasspathFingerprintCompareStrategyTest.groovy

```diff
@@ -21,224 +21,163 @@ import com.google.common.collect.Iterables
             [:]
-        ) as List == results
-
-        where:
-        strategy     | includeAdded | results
-        CLASSPATH    | true         | [added("one-new": "one")]
-        CLASSPATH    | false        | []
+        ) as List == [added("one-new": "one")]
     }
 
-    @Unroll
-    def "non-trivial addition (#strategy, include added: #includeAdded)"() {
+    def "non-trivial addition"() {
         expect:
-        changes(strategy, includeAdded,
+        changes(
             ["one-new": fingerprint("one"), "two-new": fingerprint("two")],
             ["one-old": fingerprint("one")]
-        ) == results
-
-        where:
-        strategy   | includeAdded | results
-        CLASSPATH  | true         | [added("two-new": "two")]
-        CLASSPATH  | false        | []
+        ) == [added("two-new": "two")]
     }
 
-    @Unroll
-    def "trivial removal (#strategy, include added: #includeAdded)"() {
+    def "trivial removal"() {
         expect:
-        changes(strategy, includeAdded,
+        changes(
             [:],
             ["one-old": fingerprint("one")]
         ) as List == [removed("one-old": "one")]
-
-        where:
-        strategy   | includeAdded
-        CLASSPATH  | true
-        CLASSPATH  | false
     }
 
-    @Unroll
-    def "non-trivial removal (#strategy, include added: #includeAdded)"() {
+    def "non-trivial removal"() {
         expect:
-        changes(strategy, includeAdded,
+        changes(
+            
             ["one-new": fingerprint("one")],
             ["one-old": fingerprint("one"), "two-old": fingerprint("two")]
         ) == [removed("two-old": "two")]
-
-        where:
-        strategy   | includeAdded
-        CLASSPATH  | true
-        CLASSPATH  | false
     }
 
-    @Unroll
-    def "non-trivial modification (#strategy, include added: #includeAdded)"() {
+    def "non-trivial modification"() {
         expect:
-        changes(strategy, includeAdded,
+        changes(
             ["one-new": fingerprint("one"), "two-new": fingerprint("two", 0x9876cafe)],
             ["one-old": fingerprint("one"), "two-old": fingerprint("two", 0xface1234)]
         ) == [modified("two-new": "two", FileType.RegularFile, FileType.RegularFile)]
-
-        where:
-        strategy   | includeAdded
-        CLASSPATH  | true
-        CLASSPATH  | false
     }
 
-    @Unroll
-    def "trivial replacement (#strategy, include added: #includeAdded)"() {
+    def "trivial replacement"() {
         expect:
-        changes(strategy, includeAdded,
+        changes(
             ["two-new": fingerprint("two")],
             ["one-old": fingerprint("one")]
-        ) as List == results
-
-        where:
-        strategy   | includeAdded | results
-        CLASSPATH  | true         | [removed("one-old": "one"), added("two-new": "two")]
-        CLASSPATH  | false        | [removed("one-old": "one")]
+        ) as List == [removed("one-old": "one"), added("two-new": "two")]
```

Comment:
`as List` shouldn't be required here.

Reply:
Removed. I wonder why it has been there in the first place...